### DISCUSSION TITLE
Best practice for reusing result of `list` request for `byId` requests

### DISCUSSION THREAD

**Discussion by PepijnSenders** [TanStack/query#1139] [Q&A]
When you do a request to for example `/characters` and then you do a request to for example `/characters/1`, is there a way to reuse the result of the list command for `/characters/1`? My use case is a bit more complex and I'm considering setting each "character" on the `onSuccess` callback of the `/characters` request. Anyone else have a better method for doing this?

**Comment by TkDodo** [comment]
You can put the result from the list query into the detail query as `initialData`:

https://react-query.tanstack.com/docs/guides/initial-query-data#initial-data-from-cache

Alternatively, you can just call `queryCache.setQueryData`:

https://react-query.tanstack.com/docs/guides/prefetching#manually-priming-a-query
